.github/workflows/unit-test.yml

```
@@ -97,61 +102,134 @@ jobs:
+          fi
+
       - name: Run PennyLane Unit Tests
+        if: steps.continue.outputs.confirm == 'true'
         env:
           PYTEST_MARKER: ${{ inputs.pytest_markers != '' && format('-m "{0}"', inputs.pytest_markers) || '' }}
         # Calling PyTest by invoking Python first as that adds the current directory to sys.path
-        run: python -m pytest  ${{ inputs.pytest_test_directory }} ${{ inputs.pytest_coverage_flags }} ${{ inputs.pytest_additional_args }} ${{ env.PYTEST_MARKER }} -n auto
+        run: python -m pytest ${{ inputs.pytest_test_directory }} ${{ steps.pytest_args.outputs.args }} ${{ env.PYTEST_MARKER }}
 
       - name: Adjust coverage file for Codecov
+        if: inputs.pipeline_mode == 'unit-tests'
         run: bash <(sed -i 's/filename=\"/filename=\"pennylane\//g' coverage.xml)
 
       - name: Upload Coverage File
+        if: inputs.pipeline_mode == 'unit-tests'
         uses: actions/upload-artifact@v3
         with:
           name: ${{ inputs.coverage_artifact_name }}
-          path: coverage.xml
\ No newline at end of file
+          path: coverage.xml
+
+        # If this is a reference benchmark move file to the cached directory, and we have a non-empty reference benchmark file.
+      - name: Move reference pytest benchmark file to cache
+        if: inputs.pipeline_mode == 'reference-benchmarks' && hashFiles('benchmarks.json') != ''
+        run: mkdir -p ${{ github.workspace }}/benchmark_reference && cp benchmarks.json "$_"
+
+      - name: Convert pytest benchmark JSON files to XUBM-JSON
+        if: inputs.pipeline_mode == 'benchmarks' && hashFiles('benchmarks.json') != '' && hashFiles('benchmark_reference/benchmarks.json') != ''
+        run: |
+          mkdir ${{ github.workspace }}/benchmark_results
+          python .github/workflows/scripts/benchmarks/convert_pytest_JSON_to_XUBM.py --filename_XUBM benchmark_results/benchmarks_xubm.json --author ${{ github.event.pull_request.user.login }} --github_reference ${{ github.ref }}
+          python .github/workflows/scripts/benchmarks/convert_pytest_JSON_to_XUBM.py --filename benchmark_reference/benchmarks.json --filename_XUBM benchmark_reference/benchmarks_xubm.json --author ${{ github.event.pull_request.user.login }}
+
+      - name: Plotting benchmark graphs
+        if: inputs.pipeline_mode == 'benchmarks' && hashFiles('benchmark_results/benchmarks_xubm.json') != '' && hashFiles('benchmark_reference/benchmarks_xubm.json') != ''
+        run: |
+          python .github/workflows/scripts/benchmarks/plot_benchmarks.py --graph_name ${{ steps.benchmark_name.outputs.benchmark_name }}
+
+      # If this is an assessment benchmark, upload the data as an artifact.
+      - name: Upload pytest benchmark results
+        if: inputs.pipeline_mode == 'benchmarks'
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ steps.benchmark_name.outputs.benchmark_name }}
```

Could we add `benchmark` as a prefix/suffix? Currently the artifact contains only the job name re-formatted, which actually does not make it clear the contents of the zip file.
```suggestion
          name: ${{ steps.benchmark_name.outputs.benchmark_name }}-benchmark
```